core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -573,7 +573,9 @@ object Chunk extends CollectorK[Chunk] with ChunkCompanionPlatform {
       if (i == 0) value else throw new IndexOutOfBoundsException()
     def copyToArray[O2 >: O](xs: Array[O2], start: Int): Unit = xs(start) = value
     protected def splitAtChunk_(n: Int): (Chunk[O], Chunk[O]) =
-      sys.error("impossible")
+      if (n == 0) (Chunk.empty, this)
```

Comment:
Hm, why do we need this? `splitAtChunk_` is only called by the `splitAtChunk` method from the super type, which is defined as:

```scala
  def splitAt(n: Int): (Chunk[O], Chunk[O]) =
    if (n <= 0) (Chunk.empty, this)
    else if (n >= size) (this, Chunk.empty)
    else splitAtChunk_(n)
```

So it appears that these new guards will never be true.

Reply:
Returned to the previous implementation